docs/core/deploying/native-aot/warnings/il3050.md

```diff
@@ -0,0 +1,36 @@
+---
+title: "IL3050: Avoid calling members annotated with 'RequiresDynamicCodeAttribute' when publishing as native AOT"
+description: "Members annotated with 'RequiresDynamicCodeAttribute' require code generation at runtime"
+ms.date: 09/01/2022
+ms.topic: reference
+ms.author: michals
+author: MichalStrehovsky
+f1_keywords:
+  - "IL3050"
+---
+# IL3050: Avoid calling members annotated with 'RequiresDynamicCodeAttribute' when publishing as native AOT
+
+## Cause
+
+When publishing an app as native AOT (for example, by setting the `PublishAot` property in a project to `true`), calling members annotated with the `RequiresDynamicCodeAttribute` attribute may result in exceptions at runtime. Members annotated with this attribute may require ability to dynamically create new code at runtime, and native AOT publishing model doesn't provide a way to generate native code at runtime.
```

Comment:
```suggestion
When publishing an app as native AOT (by setting the `PublishAot` property in a project to `true`), calling members annotated with the `RequiresDynamicCodeAttribute` attribute may result in exceptions at runtime. Members annotated with this attribute may require ability to dynamically create new code at runtime, and native AOT publishing model doesn't provide a way to generate native code at runtime.
```
Setting the PublishAot property is the only way to publish as native aot now.

Reply:
When referring to the time during which an app runs, we have settled on "run time" for the docs.

Reply:
```suggestion
When publishing an app as native AOT (by setting the `PublishAot` property in a project to `true`), calling members annotated with the `RequiresDynamicCodeAttribute` attribute may result in exceptions at run time. Members annotated with this attribute may require ability to dynamically create new code at run time, and native AOT publishing model doesn't provide a way to generate native code at run time.
```